test/suite/test_hs33.py

```diff
@@ -0,0 +1,150 @@
+            session.begin_transaction()
+            cursor.set_key(ds.key(i))
+
+            mods = []
+            mod = wiredtiger.Modify('A', offset, 1)
+            mods.append(mod)
+            self.assertEqual(cursor.modify(mods), 0)
+
+            if timestamp == True:
+                session.commit_transaction('commit_timestamp=' + self.timestamp_str(i + 1))
+            else:
+                session.commit_transaction()
+        cursor.close()
+
+    def add_insert(self, uri, ds, value, nrows):
+        session = self.session
+        cursor = session.open_cursor(uri)
+        self.pr('insert: ' + uri + ' for ' + str(nrows) + ' rows')
+        for i in range(1, nrows):
+            session.begin_transaction()
+            cursor[ds.key(i)] = value
+            session.commit_transaction()
+        cursor.close()
+
+    def test_hs_recovery(self):
+        ntables = 100
+        nrows = 5
+
+        bigvalue = b"aaaaa" * 100
+        bigvalue2 = b"ccccc" * 100
+
+        # Create a large number of tables.
+        tables = {}
+        for i in range(1, ntables):
+            uri = 'table:table' + str(i)
+            ds = SimpleDataSet(self, uri, 0, key_format='S', value_format='u',
+                               config='log=(enabled=false)')
+            ds.populate()
+            tables[uri] = ds
+
+        # Insert an initial set of data.
+        for uri, ds in tables.items():
+            self.add_insert(uri, ds, bigvalue, nrows)
+
+        # Apply some updates to move data to the history store.
+        session2 = self.conn.open_session()
+        session2.begin_transaction()
+        # Large updates with session 1.
+        for uri, ds in tables.items():
+            self.large_updates(self.session, uri, bigvalue2, ds, nrows)
+
+        self.session.checkpoint()
+        session2.rollback_transaction()
+        session2.close()
+
+        self.conn.reconfigure('timing_stress_for_test=[checkpoint_stop]')
+
+        # Create a checkpoint thread
+        done = threading.Event()
+        ckpt = checkpoint_thread(self.conn, done)
+        try:
+            ckpt.start()
+
+            # Wait until we reach the checkpoint stop timing stress point before copying the
+            # database. This point is placed before we sync the metadata file so that the snapshot
+            # includes incomplete checkpoint log records.
+            ckpt_stop_timing_stress = 0
+            while not ckpt_stop_timing_stress:
+                time.sleep(1)
+                stat_cursor = self.session.open_cursor('statistics:', None, None)
+                ckpt_stop_timing_stress = stat_cursor[stat.conn.checkpoint_stop_stress_active][2]
+                stat_cursor.close()
+
+            copy_wiredtiger_home(self, '.', "RESTART")
+
+        finally:
+            done.set()
+            ckpt.join()
+
+        # Open the new directory, triggering recovery. Set a low eviction size and triggers to
+        # trigger the eviction checks.
```

Comment:
Can we expand here and say that before the newly added fix, we would end up opening the HS as part of the eviction flow? Now, since we are preventing this from happening, it is no longer an issue.

Reply:
Updated https://github.com/wiredtiger/wiredtiger/pull/11857/commits/99c766012f9bb54cc3c8ab7bc6ef02548a2d635e